gamestonk_terminal/cryptocurrency/crypto_controller.py

```diff
@@ -117,21 +123,64 @@ def call_book(self, other_args):
         """Process book command"""
         binance_model.order_book(other_args, self.current_coin, self.current_currency)
 
-    def call_candle(self, other_args):
+    def call_candle(self, _):
         """Process candle command"""
-        binance_model.show_candles(other_args, self.current_coin, self.current_currency)
+        binance_model.show_candles(
+            self.current_df, self.current_coin, self.current_currency
+        )
 
     def call_balance(self, _):
         """Process balance command"""
         binance_model.balance(self.current_coin)
 
     def call_select(self, other_args):
         """Process select command"""
-        self.current_coin, self.current_currency = binance_model.select_binance_coin(
-            other_args
-        )
+        (
+            self.current_coin,
+            self.current_currency,
+            self.current_df,
+        ) = binance_model.select_binance_coin(other_args)
+        self.source = "BIN"
         print("")
 
+    # pylint: disable=inconsistent-return-statements
+    def call_ta(self, _):
+        """Process ta command"""
+        if not self.current_coin:
+            print("Please load a coin through either load or select", "\n")
+        if self.current_df.empty:
+            print("Price dataframe is empty")
+        else:
+            # Need to make the columns in df be compatible.  Also since there are no splits or dividends, there is no
+            # adj close.  To tell ta to use Close, we can just set the interval to anything but 1440.
+            # Binance provides candles so we just need to rename:
+            if self.source == "BIN":
+                self.current_df = self.current_df.rename(
+                    columns={
+                        "Open": "1. open",
+                        "High": "2. high",
+                        "Low": "3. low",
+                        "Close": "4. close",
+                        "Volume": "6. volume",
+                    }
+                )
+                self.current_df.index.name = "date"
+
+            # Coingecko does not provide candles so we can only provide close data.
+            elif self.source == "CG":
+
+                self.current_df = self.current_df[["Price"]].rename(
+                    columns={"Price": "4. close"}
+                )
+                self.current_df.index.name = "date"
+
+            return ta_controller.menu(
+                self.current_df,
+                self.current_coin,
+                self.current_df.index[0],
+                "Hi",
```

Comment:
This may be due to your comment above, right?

Reply:
The `hi`?  That can be anything except "1440" so that ta knows to not use adjusted close